code/clue.js

```diff
@@ -1,59 +1,464 @@
+  };
+  return result;
+};
+
+const pickFavouriteWeapon = (suspect) => {
+  const newArrayOfWeapons = weapons.map(a => a.name);
+  const randomWeapons = [
+    ...newArrayOfWeapons,
+    ...favouriteWeaponDuplicated(suspect.favouriteWeapon, 10)
+  ];
+  mystery.pickedWeapon = randomSelector(randomWeapons);
+  const mysteryWeapon = weapons.find((item) => {
+    if (item.name === mystery.pickedWeapon) {
+      return item;
+    };
+  });
+  mystery.weapon = mysteryWeapon;
+  mystery.favouriteRoom = mysteryWeapon.favouriteRoom;
+  weaponLoader.style.display = "none";
+  weaponCard.style.background = "#808080";
+  weaponName.innerHTML = mystery.weapon.name;
+  weaponWeight.innerHTML = `Weight: ${mystery.weapon.weight} oz`;
+  weaponImage.style.display = "inline-block";
+  weaponImage.src = mystery.weapon.image;
+  weaponFavouriteRoom.innerHTML = `Location: ${mystery.weapon.favouriteRoom}`;
+};
+
+weaponCard.onclick = favouriteWeaponAnimation;
+
+// Actions related to Room deck
+
+const favouriteRoomAnimation = () => {
+  if (mystery.killer === undefined) {
+    revealedMystery.innerHTML = ('Pick a killer first!');
+  } else if (mystery.weapon === undefined) {
+    revealedMystery.innerHTML = ('Pick a weapon first!');
+  } else {
+    roomLoader.style.display = "block";
+    roomName.innerHTML = '';
+    roomImage.src = '';
+    roomImage.style.display = "none";
+    setTimeout(function () {
+      pickFavouriteRoom(mystery.weapon);
+    }, 2000);
+  };
+};
+
+const favouriteRoomDuplicated = (elementToPopulate, howManyTimes) => {
+  let result = [];
+  for (let i = 0; i < howManyTimes; i++) {
+    result.push(elementToPopulate);
+  };
+  return result;
+};
+
+const pickFavouriteRoom = (weapon) => {
+  const newArrayOfRooms = rooms.map(a => a.name);
+  console.log(newArrayOfRooms);
+  const randomRooms = [
+    ...newArrayOfRooms,
+    ...favouriteRoomDuplicated(weapon.favouriteRoom, 30)
+  ];
+  console.log(favouriteRoomDuplicated(weapon.favouriteRoom, 30));
+  console.log(randomRooms);
+  mystery.pickedRoom = randomSelector(randomRooms);
+  const mysteryRoom = rooms.find((item) => {
+    if (item.name === mystery.pickedRoom) {
+      return item;
+    };
+  });
+  console.log(mysteryRoom);
+  mystery.room = mysteryRoom;
+  roomLoader.style.display = "none";
+  roomCard.style.background = "#C0C0C0";
+  roomName.innerHTML = mystery.room.name;
+  roomImage.style.display = "inline-block";
+  roomImage.src = mystery.room.image;
+};
+
+roomCard.onclick = favouriteRoomAnimation;
+
+// Functions related to revelation of mystery 
+
+const revealMystery = () => {
+  if (mystery.killer === undefined || mystery.weapon === undefined || mystery.room === undefined) {
+    revealedMystery.innerHTML = ('Make sure you have picked a killer, a weapon and a room!');
+  } else {
+    revealedMystery.innerHTML = (`The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName} with the ${mystery.weapon.name} in the ${mystery.room.name}.`);
+  }
+};
```

Comment:
Really nice that you made sure the user gets prompted to pick cards and to pick the killer first! 😎

Reply:
Could also be written the other way around:
```
if (myster.killer && mystery.weapon && mystery.room) {
  //Reveal the mystery
} else {
  //Pick some cards first
}
```